liblumen_alloc/src/borrow/clone_to_process.rs

```diff
@@ -53,6 +55,6 @@ pub trait CloneToProcess {
 
     /// Returns the size in words needed to allocate this value
     fn size_in_words(&self) -> usize {
-        erts::to_word_size(mem::size_of_val(self))
+        erts::to_word_size(Layout::for_value(self).size())
```

Comment:
* I assume this is more correct because it handles overage for packing/alignment?
* Did using `mem::size_of_val` cause actual segfaults?

Reply:
Correct, `Layout::for_value` adds padding for alignment, and its major use case is for the dynamically-sized types (`Tuple/HeapBin/Closure`) as calculating the correct size/alignment/padding is error-prone to do manually, and especially for those.

I suspect that our memory bugs/segfaults are almost entirely due to incorrect assumptions about behavior regarding padding and alignment; or performing calculatings for layout differently in various places - I've been consolidating on using `Layout` everywhere and deprecating use of the lower-level primitives, except in cases where it either isn't being used in memory management, or is necessary. There are still plenty of places using `mem::size_of_val` or `mem::size_of`, but I'm planning to make any of those usages around memory management go away in favor of `Layout`